gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -721,65 +879,76 @@ public void shouldRedirectToMFAChallengePage_adaptiveMFA() throws Exception {
     @Test
     public void shouldRedirectToMFAChallengePage_adaptiveMFA_with_step_up_true_strong_auth_true() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String challengeRule = "{#context.attributes['geoip']['country_iso_code'] == 'FR'}";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.RISK_BASED, challengeRule);
```

Comment:
**todo** : based on the test case name, the test is related to adaptive MFA but the Challenge type is RISK_BASED, should be CONDITIONAL